### ISSUE TITLE
Behavior of the mdl-createmodel command

### ISSUE THREAD

**Issue by aabest** [ModelSEED/Model-SEED-core#164]
I have started to generate models in my local ModelSEED rather than submitting to the web version of the ModelSEED, since that service seems to be slow and potentially unreliable in returning completed jobs. I can successfully create models using the mdl-createmodel command, but I have some questions about what is occurring.
1. Is this the correct command to be using to generate new models from previously annotated genomes?
2. The output to the terminal after running the command gives quite a bit of information about the model creation process including how the particular genome is meeting certain criteria used for generating the biomass equation.
   
   a. Is this information recorded somewhere other than the terminal output? It might be nice to be able to get back to that information in a log file or something.
   
   b. I am noticing at least one specific issue with the answers to the criteria... in particular whether the organism is Gram negative or Gram positive. I believe this will impact what cell wall and lipid components get automatically put into the biomass. I am including terminal output for one of my models. It is a Shewanella species, which is Gram negative... however, you'll notice that it is not classified as such. Where is that determined and what is the impact of getting it wrong?

---

best@Delft:~/Model-SEED-core/workspace/AaronB/Shew5$mdl-createmodel 326297.12?Seed326297.12?0?AaronB??1?1?1
mdlcreatemodel
AND{SUBSYSTEM:One-carbon_metabolism_by_tetrahydropterines|SUBSYSTEM:Folate_Biosynthesis|!SUBSYSTEM:One-carbon_metabolism_by_tetrahydropterines`H} : YES
AND{SUBSYSTEM:One-carbon_metabolism_by_tetrahydropterines|SUBSYSTEM:Folate_Biosynthesis|!SUBSYSTEM:One-carbon_metabolism_by_tetrahydropterines`H} : YES
AND{SUBSYSTEM:One-carbon_metabolism_by_tetrahydropterines|SUBSYSTEM:Folate_Biosynthesis|!SUBSYSTEM:One-carbon_metabolism_by_tetrahydropterines`H} : YES
OR{SUBSYSTEM:Glutathione:_Biosynthesis_and_gamma-glutamyl_cycle`A`B|SUBSYSTEM:Glutathione:_Non-redox_reactions`A|SUBSYSTEM:Glutathione:_Redox_cycle`A`B} : NO
AND{SUBSYSTEM:Heme_and_Siroheme_Biosynthesis`A`B`F} : NO
AND{SUBSYSTEM:Heme_and_Siroheme_Biosynthesis`A`F} : NO
AND{SUBSYSTEM:Polyamine_Metabolism} : YES
AND{SUBSYSTEM:Polyamine_Metabolism`A`B`C`D`E`F`G} : NO
AND{SUBSYSTEM:Thiamin_biosynthesis} : YES
AND{SUBSYSTEM:Ubiquinone_Biosynthesis} : YES
AND{SUBSYSTEM:Menaquinone_and_Phylloquinone_Biosynthesis} : YES
AND{SUBSYSTEM:Menaquinone_and_Phylloquinone_Biosynthesis|ROLE:Ubiquinone/menaquinone biosynthesis methyltransferase UbiE (EC 2.1.1.-)} : YES
AND{SUBSYSTEM:Coenzyme_B12_biosynthesis} : YES
AND{ROLE:Cardiolipin synthetase (EC 2.7.8.-)|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{ROLE:Cardiolipin synthetase (EC 2.7.8.-)|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{ROLE:Cardiolipin synthetase (EC 2.7.8.-)|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{OR{ROLE:Phosphatidylglycerophosphatase B (EC 3.1.3.27)|ROLE:Phosphatidylglycerophosphatase A (EC 3.1.3.27)}|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : AND{YES|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII}
AND{YES|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{OR{ROLE:Phosphatidylglycerophosphatase B (EC 3.1.3.27)|ROLE:Phosphatidylglycerophosphatase A (EC 3.1.3.27)}|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : AND{YES|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII}
AND{YES|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{OR{ROLE:Phosphatidylglycerophosphatase B (EC 3.1.3.27)|ROLE:Phosphatidylglycerophosphatase A (EC 3.1.3.27)}|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : AND{YES|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII}
AND{YES|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{ROLE:Phosphatidylserine decarboxylase (EC 4.1.1.65)|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{ROLE:Phosphatidylserine decarboxylase (EC 4.1.1.65)|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{ROLE:Phosphatidylserine decarboxylase (EC 4.1.1.65)|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : YES
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive|SUBSYSTEM:Fatty_Acid_Biosynthesis_FASII} : NO
AND{CLASS:Gram positive} : NO
AND{CLASS:Gram negative} : NO
AND{!NAME:Mycoplasma|!NAME:Spiroplasma|!NAME:Ureaplasma|!NAME:phytoplasma} : YES
AND{!NAME:Mycoplasma|!NAME:Spiroplasma|!NAME:Ureaplasma|!NAME:phytoplasma} : YES
OR{COMPOUND:cpd15665|COMPOUND:cpd15667|COMPOUND:cpd15668|COMPOUND:cpd15669} : YES
AND{COMPOUND:cpd00166} : YES
AND{COMPOUND:cpd00166} : YES
Total mass = -1.11299999999999, Reactant mass = -22.1950879901449, Product mass = 21.0820879901449
NOMATCH!
Creating problem directory: 35
Creating list of inactive reactions: /Users/best/Model-SEED-core/data/ReactionDB/MFAToolkitOutputFiles/35/InactiveModelReactions.txt
Printing gapfilling parameters in : /Users/best/Model-SEED-core/data/ReactionDB/MFAToolkitOutputFiles/35/CompleteGapfillingParameters.txt
Running runFBAStudy
Using problem directory: 35
Creating the model file: /Users/best/Model-SEED-core/data/ReactionDB/MFAToolkitOutputFiles/35/Seed326297.12.906.tbl
Creating biochemistry provenance files
Running FBA
mkdir: /Users/best/Model-SEED-core/data/ReactionDB/MFAToolkitOutputFiles/35/MFAOutput: File exists
IBM ILOG License Manager: "IBM ILOG Optimization Suite for Academic Initiative" is accessing CPLEX 12 with option(s): "e m b q ".
drncpd11416 FAILED  NONE    Prelim  23  --
drncpd15302 FAILED  NONE    Prelim  23  --
drncpd08636 FAILED  NONE    Prelim  23  --
bio08958    +rxn05149;+rxn05039;+rxn05722;+rxn03891;+rxn05433;+rxn05437;+rxn05436;+rxn05435;+rxn05434;+rxn05440;+rxn05439;+rxn05438;+rxn05442;+rxn05441;+rxn05444;+rxn05443;+rxn03394;+rxn08131;+rxn09345;+rxn10447;-rxn12008;+rxn13477     1/1 60  0
Parsing results
Finished runFBAStudy
rxn12008    <=
rxn05722    =>
rxn05435    =>
rxn03394    =>
rxn03891    =>
rxn05433    =>
rxn05437    =>
rxn05439    =>
rxn13477    =>
rxn05434    =>
rxn08131    =>
rxn05441    =>
rxn05444    =>
rxn05149    =>
rxn05443    =>
rxn05039    =>
rxn05438    =>
rxn05440    =>
rxn05442    =>
rxn09345    =>
rxn10447    =>
rxn05436    =>
Deleting previously auto-completed reaction: rxn09680
Adding reaction rxn12008
Adding reaction rxn03891
Adding reaction rxn13477
Adding reaction rxn08131
Adding reaction rxn09345
Calculting the growth with which to test the model
Using problem directory: 36
Creating the model file: /Users/best/Model-SEED-core/data/ReactionDB/MFAToolkitOutputFiles/36/Seed326297.12.906.tbl
Creating biochemistry provenance files
Running FBA
mkdir: /Users/best/Model-SEED-core/data/ReactionDB/MFAToolkitOutputFiles/36/MFAOutput: File exists
Parsing results
Successfully created model Seed326297.12.906!

**Comment by cshenry** [comment]
Aaron, you should get the same result from using the web submission or the local Model SEED. Both systems use the same code. I apologize for the poor reliability of the web system. We had a number of problems with our queue throughout May and part of June. We have resolved these now, and we've put tests in place to ensure that we are notified immediately should the queue break down in the future.

Anyhow, you are using the right command. Sadly, the output to the screen is not saved anywhere. You can of course save this text yourself by piping it to a file using " > CreateModelOutput.txt" at the end of the mdl-createmodel command.

The biomass template does change depending on whether the organism is gram positive or gram negative, and the system does sometimes get the calls wrong. In these cases, the biomass reaction will have the wrong biomass. Fortunately, ModelSEED has a mechanism for over-riding the call and forcing the system to consider a particular genome as "gram negative" or "gram positive". Open this file: "Model-SEED-core/lib/ModelSEED/FIGMODELconfig.txt", and find the lines that start with: "%Gram positive" and "%Gram negative". Add your genome to the appropriate list. Make the same change to this file: "Model-SEED-core/config/FIGMODELconfig.txt", or just run a ms-config.pl. Note, you have to change both files to avoid overwriting your change when you run ms-config.pl.

Once you've made this change, you should be able to run the mdl-reconstruction command for your genome. 

Note, I did need to make a small tweak in the code to make this work, so you should do a "git pull" before doing all this.

**Comment by aabest** [comment]
Thanks for the workaround, Chris. While doing the code modifications, I noticed that there were lines for "families" where it looks like you are pattern matching on genus names. I modified that line to include the genus, since I'll be building many models for this genus.

This brings up another question though.. I tried building another model (so as not to overwrite the current model I have) and compare to the existing one using the mdl-createmodel command. This did not recognize that I had included the genus name in the families line. However, I followed that up by running the mdl-reconstruction on the newly generated model, and it did recognize the genus name and classified the organism correctly. Is this how this is supposed to behave? Does the code for the mdl-createmodel command need to be updated, as well?
